vizro-core/tests/tests_utils/asserts.py

```diff
@@ -4,27 +4,31 @@
 import plotly
 
 
-def strip_keys(object, keys):
-    """Strips all entries with key "id" from a dictionary, regardless of how deeply it's nested.
-
-    This makes it easy to compare dictionaries generated from Dash components we've created that contain random IDs.
-    """
+def _strip_keys(object, keys):
+    """Strips from a JSON object all entries where the key is in keys, regardless of how deeply it's nested."""
     if isinstance(object, dict):
-        object = {key: strip_keys(value, keys) for key, value in object.items() if key not in keys}
+        object = {key: _strip_keys(value, keys) for key, value in object.items() if key not in keys}
     elif isinstance(object, list):
-        object = [strip_keys(item, keys) for item in object]
+        object = [_strip_keys(item, keys) for item in object]
     return object
 
 
-def component_to_dict(component: dash.development.base_component.Component) -> dict:
+def _component_to_dict(component: dash.development.base_component.Component) -> dict:
+    """Prepares a Dash component for comparison by conversion to JSON object."""
     return json.loads(json.dumps(component, cls=plotly.utils.PlotlyJSONEncoder))
 
 
-# TODO: implement some sort of depth limit to comparison so can use in high level tests, roll out more widely across
-# tests
 def assert_component_equal(left, right, keys_to_strip=None):
-    # Note we check for None explicitly because {} is a valid value for keys_to_strip.
-    keys_to_strip = keys_to_strip if keys_to_strip is not None else {"id", "class_name", "className"}
-    left = strip_keys(component_to_dict(left), keys_to_strip)
-    right = strip_keys(component_to_dict(right), keys_to_strip)
+    """Checks that the left and right Dash components are equal, ignoring keys_to_strip.
+
+    Examples:
+        >>> from dash import html
+        >>> assert_component_equal(html.Div(), html.Div())
+        >>> assert_component_equal(html.Div(id="a"), html.Div(), keys_to_strip={"id"})
+        >>> assert_component_equal(html.Div(html.Div()), html.Div(), keys_to_strip={"children"})
+    """
+    keys_to_strip = keys_to_strip or {}
+
+    left = _strip_keys(_component_to_dict(left), keys_to_strip)
+    right = _strip_keys(_component_to_dict(right), keys_to_strip)
```

Comment:
The tests would be faster if we first `strip (remove) the object attributes` (similar to `_strip_keys` in dict) and then `translate the "truncated" object into a dictionary`. If you find it easy to implement, I'd love to see it in a PR, otherwise ignore this comment 😄.

Reply:
Please could you explain a bit more what you mean here? This is what a component looks like before conversion to dictionary:

```
component = Div(children=Button(children='My text', id='button_id', className='button_primary'), id='button_id_outer', className='button_container')
```

Each Dash component has an attribute `available_properties` and `children`, so I guess something like:

```
for key in keys_to_strip:
    remove component.available_properties[key]
    # recurse through component.children and do the same
```

Is that what you mean?

Reply:
Yes, that's what I meant.

So my assumptions are that this line:
`_component_to_dict(_strip_component_properties(dash_component, properties_to_strip))`

is faster for bulky and nested dash components than this one:
`_strip_keys(_component_to_dict(dash_component), keys_to_strip)`

I can't guess how much time we can save for the `hatch run test-unit', but it might be worth a try.

Reply:
I think you're _probably_ right, although it's not obvious that would be faster actually because it would still need to be recursive code either way.

I played around with this for a while and couldn't quickly find a simple way to do it though. I did check performance of the current solution, and given that our nested dictionaries are never that big the time optimisation here is going to be extremely minimal even if `assert_component_equal` is used 10000s of times across the test suite. There are much bigger overheads in running the test suite than this at the moment, so even if we could speed the check up by a factor of 100 it would have minimal impact on overall running time.

Reply:
So definitely a good suggestion, but I'm going to leave it how it is 🙂